### PR TITLE
Merge repos one virtualmesh per mesh

### PR DESCRIPTION
**Pull request by harveyxia**

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -9,6 +9,8 @@ jobs:
   build:
     machine:
       image: ubuntu-1604:201903-01
+      # need to upgrade plan before using large executor
+      # resource_class: large
```
Comment (joekelley) [inline .circleci/config.yml:13]: do we need this upgrade? Looks like CI is green as-is.

File: .github/workflows/pull_request.yaml [hunk 2]
```diff
@@ -25,6 +25,13 @@ jobs:
           key: ${{ runner.os }}-go-${{ hashFiles('**/go.sum') }}
           restore-keys: |
             ${{ runner.os }}-go-
+      - name: Install Protoc
+        uses: solo-io/setup-protoc@master
+        with:
+          version: '3.6.1'
+      - name: Check code gen
+        run: |
+          ./ci/check-code-gen.sh
```
Comment (joekelley) [inline .github/workflows/pull_request.yaml:34]: In other projects we've made this a separate job from unit tests. That way we get better parallelization and a better bird-eye-view of why CI is failing.

File: api/networking/v1alpha2/virtual_mesh.proto [hunk 3]
```diff
@@ -149,7 +150,7 @@ message VirtualMeshStatus {
     // the state of the overall resource.
     // will only show accepted if it has been successfully
     // applied to all target meshes.
-    ApprovalState state = 2;
+    ApprovalStatus status = 2;
 
     // The status of the VirtualMesh for each Mesh to which it has been applied.
     // A TrafficPolicy may be Accepted for some Meshes and rejected for others.
```
Comment (joekelley) [inline api/networking/v1alpha2/virtual_mesh.proto:156]: While we're here, this comment seems out of place

File: api/networking/v1alpha2/virtual_mesh.proto [hunk 4]
```diff
@@ -149,7 +150,7 @@ message VirtualMeshStatus {
     // the state of the overall resource.
     // will only show accepted if it has been successfully
     // applied to all target meshes.
```
Comment (joekelley) [inline api/networking/v1alpha2/virtual_mesh.proto:152]: nit: let's update this comment to describe status rather than state

File: pkg/common/bootstrap/bootstrap.go [hunk 5]
```diff
@@ -20,12 +20,17 @@ import (
 	_ "k8s.io/client-go/plugin/pkg/client/auth"
 )
 
+type StartParameters struct {
```
Comment (joekelley) [inline pkg/common/bootstrap/bootstrap.go:23]: mega-nit: I see structs named like "StartOptions" or "StartOpts" in many of the kube/controller libraries we interact with. In this case though, many of these seem to be required, so perhaps "Options" is misleading. wdyt?

File: .circleci/config.yml (hunk 1 above)
Comment (harveyxia) [reply .circleci/config.yml:13]: I just left this here for reference in case we ever need/want to use larger machines

File: api/networking/v1alpha2/virtual_mesh.proto [hunk 6]
```diff
@@ -141,18 +142,17 @@ message VirtualMeshSpec {
...
+    // If the observedGeneration does not match generation, the controller has not received the most
     // recent version of this resource.
     int64 observed_generation = 1;
 
-    // the state of the overall resource.
-    // will only show accepted if it has been successfully
+    // The status of the overall resource. It will only show accepted if it has been successfully
     // applied to all target meshes.
-    ApprovalState state = 2;
+    ApprovalStatus status = 2;
```
Comment (ilackarms) [inline api/networking/v1alpha2/virtual_mesh.proto:152]: why this rename?

Comment (ilackarms) [reply api/networking/v1alpha2/virtual_mesh.proto:152]: the convention i've been adhering to is as follows: 
- `status` represents the kubernetes subresource `status` which aggregates all information about the current state of the object. 
- `state` represents the current *state* of the object as an enum. 

i'd like to try to avoid overloading the term `status` if possible.

File: pkg/mesh-networking/approval/approver_test.go [hunk 7]
```diff
@@ -5,14 +5,15 @@ import (
...
+	corev1 "github.com/solo-io/skv2/pkg/api/core.skv2.solo.io/v1"
 
 	. "github.com/onsi/ginkgo"
 	. "github.com/onsi/gomega"
 	discoveryv1alpha2 "github.com/solo-io/service-mesh-hub/pkg/api/discovery.smh.solo.io/v1alpha2"
 	discoveryv1alpha2sets "github.com/solo-io/service-mesh-hub/pkg/api/discovery.smh.solo.io/v1alpha2/sets"
 	"github.com/solo-io/service-mesh-hub/pkg/api/networking.smh.solo.io/input"
 	"github.com/solo-io/service-mesh-hub/pkg/api/networking.smh.solo.io/output"
-	"github.com/solo-io/service-mesh-hub/pkg/api/networking.smh.solo.io/v1alpha2"
+	networkingv1alpha2 "github.com/solo-io/service-mesh-hub/pkg/api/networking.smh.solo.io/v1alpha2"
```
Comment (ilackarms) [inline pkg/mesh-networking/approval/approver_test.go:16]: why this change? 

in `discovery`, the discovery api is called `v1alpha2`, because it's the "local" api group.

the same thing is true in `networking`.

File: api/networking/v1alpha2/virtual_mesh.proto (hunk 6 above)
Comment (harveyxia) [reply api/networking/v1alpha2/virtual_mesh.proto]: I thought we needed AcceptanceOrder, but turns out we don't— @ilackarms makes the point that since the algorithm has the invariant of never permitting overlapping VirtualMeshes to be accepted.
